File: algo-expert/python/algo_expert/strings/easy/semordnilap.py
def semordnilap(words: list) -> list:
    result = []
    for i in range(len(words) - 1):
        palindrome = words[i][::-1]
        for j in range(i + 1, len(words)):
            if palindrome == words[j]:
                result.append([words[i], words[j]])
    return result

File: algo-expert/typescript/src/strings/easy/Semordnilap.ts
export function semordnilap(words: string[]) {
  let palindromes = new Map<string, string>();
  for (let i = 0; i < words.length - 1; i++) {
    if (palindromes.get(words[i]) != undefined) {
      continue;
    }
    for (let j = i + 1; j < words.length; j++) {
      let reverse = "";
      for (let char of words[j]) {
        reverse = char.concat(reverse);
      }
      if (words[i] === reverse) {
        palindromes.set(words[j], words[i]);
      }
    }
  }
  let result = [];
  const iterator = palindromes.keys();
  let key;
  while (!(key = iterator.next()).done) {
    result.push([palindromes.get(key.value), key.value])
  }
  return result;
}
